src/plot_api/helpers.js

```diff
@@ -217,7 +217,7 @@ exports.cleanLayout = function(layout) {
 function cleanAxRef(container, attr) {
     var valIn = container[attr];
     var axLetter = attr.charAt(0);
-    if(valIn && valIn !== 'paper') {
+    if(valIn && valIn !== 'paper' && !valIn.endsWith('domain')) {
```

Comment:
I don't believe we polyfill `endsWith` (for IE11) so we shouldn't use it - here and a few other places too.

Also in principle we should extend `cleanId` to support domain references - so `'x1 domain'` would get converted to `'x domain'`. This would be best as an option to `cleanId` since domain references are only allowed in certain places.

Reply:
Ok :+1: It looks like the place where `cleanId` could see axis IDs ending with ` domain` is in the places where `cleanAxRef` is used in `src/plot_api/helpers.js`, so I've added an argument to `cleanId` for allowing "domain" type axis references, and set this argument when calling `cleanId` in `cleanAxRef`.
But I noticed that cleanAxRef is not used for images? ... confirming this ...

Reply:
Yes, indeed if you set a funny axis reference like xref: 'x1' for images, the image doesn't draw. So I've added `cleanAxRef` for images into `src/plot_api/helpers.js` by following what is done there for shapes.

Reply:
:+1: